protocols/imap/src/main/java/org/apache/james/imap/processor/IdProcessor.java

```diff
@@ -83,7 +84,10 @@ private Mono<Void> logMailUserAgent(String mailUserAgent) {
     private void addMailUserAgentToMDC(ImapSession session, String mailUserAgent) {
         Object maybeMDC = session.getAttribute(MDC_KEY);
         if (maybeMDC instanceof MDCBuilder boundMDC) {
-            MDCBuilder newMDC = boundMDC.addToContext("mailUserAgent", mailUserAgent);
+            if (boundMDC.snapshotContextMap().containsKey(MAIL_USER_AGENT_KEY)) {
```

Comment:
Materialising the map here is uggly. 

Can't we rather rely on the recording of the MUA done in the session?